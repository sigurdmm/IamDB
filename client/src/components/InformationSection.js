import React from 'react';
import PropTypes from 'prop-types';
import './InformationSection.less';

const defaultImg = 'https://nti.biz/globalassets/images/cards-286x160px/no-image-found.jpg';

const InformationSection = ({ details }) => (
  <div className='information__container'>
        <MediaImage image={details.thumbnails ? details.thumbnails.small : defaultImg }/>
    <div className='details__container'>
        <Title title={details.name}/>
        <ImdbRating rating={`imdb rating: ${details.rating}`}/>
        <MediaType type={`Type of media: ${details.type}`}/>
        <MediaDirector director={`Director: ${details.director}`}/>
        </div>
    </div>
);

InformationSection.propTypes = {
  details: PropTypes.shape.isRequired,
};

const MediaImage = ({ image }) => <div className='image__container'><img className='image__item' src={image}/></div>;

MediaImage.propTypes = {
  image: PropTypes.string.isRequired,
};

const Title = ({ title }) => <div className='title__container'><h2>{title}</h2></div>;

Title.propTypes = {
  title: PropTypes.string.isRequired,
};

const ImdbRating = ({ rating }) => <p>{rating}</p>;

ImdbRating.propTypes = {
  rating: PropTypes.number,
};

const MediaType = ({ type }) => <p>{type}</p>;

MediaType.propTypes = {
  type: PropTypes.string,
};

const MediaDirector = ({ director }) => <p>{director}</p>;

MediaDirector.propTypes = {
  director: PropTypes.string,
};

export default InformationSection;
