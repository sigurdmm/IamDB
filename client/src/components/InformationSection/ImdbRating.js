import PropTypes from 'prop-types';
import React from 'react';

const ImdbRating = ({ rating }) => <h3>{`Imdb rating: ${rating}`}</h3>;

ImdbRating.propTypes = {
  rating: PropTypes.string,
};

ImdbRating.defaultProps = {
  rating: 'Unknown',
};

export default ImdbRating;
