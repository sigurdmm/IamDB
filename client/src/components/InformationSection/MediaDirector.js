import PropTypes from 'prop-types';
import React from 'react';

const MediaDirector = ({ director }) => <h3>{`ðŸŽ¬: ${director || 'unknown'}`}</h3>;

MediaDirector.propTypes = {
  director: PropTypes.string,
};

export default MediaDirector;
