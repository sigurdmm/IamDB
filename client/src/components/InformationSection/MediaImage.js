import PropTypes from 'prop-types';
import React from 'react';
import defaultImage from '../../../public/no-image-found.jpg';

const MediaImage = ({ image }) => <div className='mediaimage'>
  <img className='image__item' src={image ? image.small : defaultImage }/>
</div>;

MediaImage.propTypes = {
  image: PropTypes.string,
};

export default MediaImage;
