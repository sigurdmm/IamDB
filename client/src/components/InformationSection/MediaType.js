import PropTypes from 'prop-types';
import React from 'react';

const MediaType = ({ type }) => <h3>Type of media: {type || 'unknown'}</h3>;

MediaType.propTypes = {
  type: PropTypes.string.isRequired,
};

export default MediaType;
