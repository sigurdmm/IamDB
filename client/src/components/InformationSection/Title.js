import PropTypes from 'prop-types';
import React from 'react';

const Title = ({ title }) => <div className='title__container'><h2>{title}</h2></div>;

Title.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Title;
