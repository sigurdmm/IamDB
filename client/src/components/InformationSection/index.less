.information__container {
  display: flex;
  flex-direction: row;

}

.title__container {
  margin-bottom: 70px;
}

.mediaimage {
  margin-right: 20px;
}

.image__item {
  height: 300px;
}


