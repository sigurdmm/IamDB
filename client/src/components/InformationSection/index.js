import React from 'react';
import PropTypes from 'prop-types';
import './index.less';
import MediaDirector from './MediaDirector';
import MediaType from './MediaType';
import ImdbRating from './ImdbRating';
import Title from './Title';
import MediaImage from './MediaImage';

const defaultImg = 'https://nti.biz/globalassets/images/cards-286x160px/no-image-found.jpg';

const InformationSection = ({ details }) => (
  <div className='information__container'>
        <MediaImage image={details.thumbnails ? details.thumbnails.small : defaultImg }/>
    <div className='details__container'>
        <Title title={details.name}/>
        <ImdbRating rating={`imdb rating: ${details.rating}`}/>
        <MediaType type={`Type of media: ${details.type}`}/>
        <MediaDirector director={`Director: ${details.director}`}/>
        </div>
    </div>
);

InformationSection.propTypes = {
  details: PropTypes.shape.isRequired,
};

export default InformationSection;
