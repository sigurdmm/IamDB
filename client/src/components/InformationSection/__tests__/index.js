import { render } from 'enzyme';

import InformationSection from '../index';

const mockedDetails = {
  thumbnails: {
    small: 'image',
  },
  name: 'title',
  rating: 8.8,
  type: 'movie',
  director: 'director',
};

describe('<InformationSection/>', () => {
  it('should render properly', () => {
    const wrapper = render(<InformationSection details={mockedDetails}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
