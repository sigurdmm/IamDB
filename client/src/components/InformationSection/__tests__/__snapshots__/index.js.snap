// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationSection/> should render properly 1`] = `
<div
  class="information__container"
>
  <div
    class="mediaimage"
  >
    <img
      class="image__item"
      src="image"
    />
  </div>
  <div>
    <div
      class="title__container"
    >
      <h2>
        title
      </h2>
    </div>
    <h3>
      ⭐️: 8.8
    </h3>
    <h3>
      🎥: movie
    </h3>
    <h3>
      🎬: director
    </h3>
  </div>
</div>
`;
